src/repl-frontend/scala/tools/nsc/interpreter/shell/ReplCompletion.scala

```diff
@@ -57,10 +57,31 @@ class ReplCompletion(intp: Repl, val accumulator: Accumulator = new Accumulator)
             case _                                               =>
               // under JLine 3, we no longer use the tabCount concept, so tabCount is always 1
               // which always gives us all completions
-              val (c, r) = result.completionCandidates(tabCount = 1)
+              val (c, r)                         = result.completionCandidates(tabCount = 1)
               // scala/bug#12238
-              val (hasDescr, noDescr) = r.partition(p => p.isDeprecated || p.isUniversal)
-              CompletionResult(buf, c, hasDescr ++ noDescr)
+              // Currently, only when all methods are Deprecated should they be displayed `Deprecated` to users. Only handle result of PresentationCompilation#toCandidates.
+              // We don't handle result of PresentationCompilation#defStringCandidates, because we need to show the deprecated here.
+              if (r.nonEmpty && r.forall(!_.defString.startsWith("def"))) {
+                val groupByDef              = r.groupBy(_.defString)
+                val allOverrideIsUniversal  = groupByDef.filter(f => f._2.size > 1 && f._2.forall(_.isUniversal)).keys.toSeq
+                val allOverrideIsDeprecated = groupByDef.filter(f => f._2.size > 1 && f._2.forall(_.isDeprecated)).keys.toSeq
+                var rewriteDescr            = List[CompletionCandidate]()
+                for (candidate <- r) {
+                  var rewriteCandidate = candidate
+                  if (allOverrideIsUniversal.contains(candidate.defString)) {
+                    rewriteCandidate = candidate.copy(isUniversal = true)
+                  } else {
+                    rewriteCandidate = candidate.copy(isUniversal = false)
+                  }
+                  if (allOverrideIsDeprecated.contains(candidate.defString)) {
+                    rewriteCandidate = candidate.copy(isDeprecated = true)
+                  } else {
+                    rewriteCandidate = candidate.copy(isDeprecated = false)
+                  }
+                  rewriteDescr = rewriteDescr.:+(rewriteCandidate)
+                }
```

Comment:
This whole segment could be replaced with

```
val rewriteSecr = r.map(candidate => {
  val isUni = allOverrideIsUniveral.contains(candidate)
  val isDepr = allOverrideIsDeprecated.contains(candidate)
  candidate.copy(isUniveral = isUni, isDeprecated = isDepr)
})
```

Reply:
good  👍